BaikalDataGenerator/Source/main.cpp

```diff
@@ -106,9 +79,40 @@ try
         return 0;
     }
 
+    if (cmd_parser.HasListDevicesOption())
+    {
+        std::cout << "Device list:\n";
+        auto devices = GetDevices();
+        for (std::size_t idx = 0; idx < devices.size(); ++idx)
+        {
+            DG_LOG(KeyValue("idx", idx)
+                << KeyValue("name", devices[idx].GetName())
+                << KeyValue("vendor", devices[idx].GetVendor())
+                << KeyValue("version", devices[idx].GetVersion()));
+        }
+        return 0;
+    }
+
+    DG_LOG(KeyValue("status", "running") << KeyValue("start_ts", std::time(nullptr)));
+
     auto config = cmd_parser.Parse();
+    ObjectLoader object_loader(config);
+    auto params = object_loader.GetDataGeneratorParams();
+
+    auto progress_callback = [](int camera_idx)
+    {
+        DG_LOG(KeyValue("event", "generated")
+            << KeyValue("status", "generating")
+            << KeyValue("camera_idx", camera_idx));
```

Comment:
camera_idx here - is index inside split, but we have to output absolute index value